server/eclipse-project/src/main/java/us/freeandfair/corla/model/ComparisonAudit.java

```diff
@@ -927,12 +927,31 @@ public OptionalInt computeDiscrepancy(final CastVoteRecord cvr,
 
   private OptionalInt computeAuditedBallotDiscrepancy(final CVRContestInfo the_cvr_info,
                                                       final CVRContestInfo the_acvr_info) {
-    // check for overvotes
+    // Check for overvotes.
+    //
+    // Overvotes are represented, perhaps confusingly, in the CVR as "all
+    // zeroes" for the given contest - it will look indistinguishable from a
+    // contest in which no selections were made. We therefore have to check if
+    // the number of selections the audit board found is less than or equal to
+    // the allowed votes for the given contest. If it is, then the audit board
+    // found a valid selection and we can proceed with the rest of the math as
+    // usual. If not, then the audit board recorded an overvote which we must
+    // now make match the way the CVR format records overvotes: we must record
+    // *no* selections. The code below does that by excluding the selections
+    // submitted by the audit board.
+    //
+    // If the CVR does show an overvote (no selections counted) then our
+    // zero-selection ACVR will match it and we will find no discrepancies. If,
+    // however, the CVR *did* show a selection but the audit board recorded an
+    // overvote, then we will be able to calculate the discrepancy - the CVR
+    // will have a choice (or choices) marked as selected, but the ACVR will
+    // not. The converse is also true: if the CVR shows an overvote but the
+    // audit board records a valid selection, we will calculate an expected
+    // discrepancy.
     final Set<String> acvr_choices = new HashSet<>();
-    // TODO: validate choices().size()
-    // if (the_acvr_info.choices().size() <= my_contest_result.votesAllowed()) {
-    acvr_choices.addAll(the_acvr_info.choices());
-    // } // else overvote so don't count the votes
+    if (the_acvr_info.choices().size() <= my_contest_result.winnersAllowed()) {
```

Comment:
@sduckett `winnersAllowed()` is presumably the same as `votesAllowed()` for our purposes, right? Votes allowed and number of candidates elected can vary independently.

Reply:
Can we use votesAllowed instead? For use in Fargo ND, where they just approved Approval Voting, the two would be different. Could we just set them equal for these CVRs, but have the code be more general?

Reply:
@nealmcb Yeah - I'm writing up a card to make that change. It's out of scope for this PR but will be needed in more general software going forward. The NIST CVR spec makes this distinction as well, which I will also link to. I'll send something your way when I'm done.

Reply:
Oh, and I can't simply rename it in this PR because the `votesAllowed` method was renamed.